tests/contrib/operators/test_livy_operator.py

```diff
@@ -0,0 +1,156 @@
+            dag=self.dag,
+            task_id='livy_example'
+        )
+        task._livy_hook = task.get_hook()
+        task.poll_for_termination(BATCH_ID)
+
+        mock_livy.assert_called_with(BATCH_ID)
+        self.assertEqual(mock_livy.call_count, 3)
+
+    @patch('airflow.contrib.operators.livy_operator.LivyHook.get_batch_state')
+    def test_poll_for_termination_fail(self, mock_livy):
+
+        state_list = 2 * [BatchState.RUNNING] + [BatchState.ERROR]
+
+        def side_effect(_):
+            if state_list:
+                return state_list.pop(0)
+            # fail if does not stop right before
+            raise AssertionError()
+
+        mock_livy.side_effect = side_effect
+
+        task = LivyOperator(
+            file='sparkapp',
+            polling_interval=1,
+            dag=self.dag,
+            task_id='livy_example'
+        )
+        task._livy_hook = task.get_hook()
+
+        with self.assertRaises(AirflowException):
+            task.poll_for_termination(BATCH_ID)
+
+        mock_livy.assert_called_with(BATCH_ID)
+        self.assertEqual(mock_livy.call_count, 3)
+
+    @patch('airflow.contrib.operators.livy_operator.LivyHook.get_batch_state',
+           return_value=BatchState.SUCCESS)
+    @patch('airflow.contrib.operators.livy_operator.LivyHook.post_batch', return_value=BATCH_ID)
+    def test_execution(self, mock_post, mock_get):
+        task = LivyOperator(
+            livy_conn_id='livyunittest',
+            file='sparkapp',
+            polling_interval=1,
+            dag=self.dag,
+            task_id='livy_example'
+        )
+        task.execute(context={})
+
+        call_args = {k: v for k, v in mock_post.call_args[1].items() if v}
+        self.assertEqual(call_args, {'file': 'sparkapp'})
+        mock_get.assert_called_once_with(BATCH_ID)
+
+    @patch('airflow.contrib.operators.livy_operator.LivyHook.delete_batch')
+    @patch('airflow.contrib.operators.livy_operator.LivyHook.post_batch', return_value=BATCH_ID)
+    def test_deletion(self, mock_post, mock_delete):
+        task = LivyOperator(
+            livy_conn_id='livyunittest',
+            file='sparkapp',
+            dag=self.dag,
+            task_id='livy_example'
+        )
+        task.execute(context={})
+        task.kill()
+
+        mock_delete.assert_called_once_with(BATCH_ID)
+
+    def test_missing_id_on_deletion(self):
+        task = LivyOperator(
+            livy_conn_id='livyunittest',
+            file='sparkapp',
+            dag=self.dag,
+            task_id='livy_example'
+        )
+
+        with self.assertRaises(AirflowException):
+            task.kill()
+
+    def test_injected_hook(self):
+        def_hook = LivyHook(livy_conn_id='livyunittest')
+
+        task = LivyOperator(
+            file='sparkapp',
+            dag=self.dag,
+            task_id='livy_example'
+        )
+        task._livy_hook = def_hook
+
+        self.assertEqual(task.get_hook(), def_hook)
+
```

Comment:
Could you add another test that checks that `.execute()` raises an AirflowError when the state isn't success please?

Reply:
that would be `test_poll_for_termination_fail`.
Jobs once submitted get scheduled for execution, so any error can be detected only when polling